### ISSUE TITLE
Nan problem in calendar

### ISSUE THREAD

**Issue by ExplosiveGM** [dangrossman/daterangepicker#2014]
1) I have empty daterangepicker
![image](https://user-images.githubusercontent.com/13543218/64188166-14299400-ce9c-11e9-822f-e1e984e822ee.png)
2) I select data range 
![image](https://user-images.githubusercontent.com/13543218/64188316-623e9780-ce9c-11e9-86a1-f0cee0bb668e.png)
3) I go to other page.
4) I click **Back** in browser and have filled datarange control from step 2

5) Now I have this problem:
![image](https://user-images.githubusercontent.com/13543218/64188785-38d23b80-ce9d-11e9-8420-bdcaf18eb461.png)

This **problem happens because** my control has **different format**.
**Inner format**:
```
  $el.daterangepicker({
    locale: {
      format: 'MM-DD-YYYY hh:mm A'
    },
  })
```
**Display format**:
```
$el.on('apply.daterangepicker', (_ev, picker) => {
    $from.val(picker.startDate.format('YYYY-MM-DD hh:mm A'))
    $to.val(picker.endDate.format('YYYY-MM-DD hh:mm A'))
    $el.closest('form').submit()
  })
```
On **step 4-5** control knows that have some data in input(code from daterangepicker.js)
```
if (typeof options.startDate === 'undefined' && typeof options.endDate === 'undefined') {
            if ($(this.element).is(':text')) {
                var val = $(this.element).val(),
                    split = val.split(this.locale.separator);

                start = end = null;

                if (split.length == 2) {
                    start = moment(split[0], this.locale.format);
                    end = moment(split[1], this.locale.format);
                } else if (this.singleDatePicker && val !== "") {
                    start = moment(val, this.locale.format);
                    end = moment(val, this.locale.format);
                }
                if (start !== null && end !== null) {
                    this.setStartDate(start);
                    this.setEndDate(end);
                }
            }
        }
```
It try 
```
start = moment(split[0], this.locale.format);
end = moment(split[1], this.locale.format);
```

In other words(it has problem with format): 
```
start = moment('2019-09-01 12:00 AM', 'MM-DD-YYYY hh:mm A')
end = moment('2019-09-05 11:59 PM',  'MM-DD-YYYY hh:mm A');
```

So i decide use the same inner and display format (''YYYY-MM-DD hh:mm A'').
But it looks like a bug.

**Comment by felipesilvati** [comment]
I'm having the same problem with Daterangepicker @3.0.5

It's only showing NaN on future Dates:

```
$('input[name="daterange"]').daterangepicker({
    opens: 'right',
    maxDate: moment().endOf('day'),
    locale: {
      format: 'DD/MM/YYYY',
      separator: ' - ',
      applyLabel: 'Confirmar',
      cancelLabel: 'Cancelar',
      fromLabel: 'De',
      toLabel: 'Até',
      customRangeLabel: 'Período Customizado',
      weekLabel: 'S',
      daysOfWeek: [
        'Dom',
        'Seg',
        'Ter',
        'Qua',
        'Qui',
        'Sex',
        'Sab'
      ],
      'monthNames': [
        'Janeiro',
        'Fevereiro',
        'Março',
        'Abril',
        'Maio',
        'Junho',
        'Julho',
        'Agosto',
        'Setembro',
        'Outubro',
        'Novembro',
        'Dezembro'
      ],
      'firstDay': 7
    },
    ranges: {
      'Hoje': [moment(), moment()],
      'Ontem': [moment().subtract(1, 'days'), moment().subtract(1, 'days')],
      'Últimos 7 dias': [moment().subtract(6, 'days'), moment()],
      'Últimos 30 dias': [moment().subtract(29, 'days'), moment()],
      'Esse mês': [moment().startOf('month'), moment().endOf('month')],
      'Último mês': [moment().subtract(1, 'month').startOf('month'), moment().subtract(1, 'month').endOf('month')]
    },
    alwaysShowCalendars: true
  }, function (start, end, label) {
    console.log('Selecionado o período: de ' + start.format('YYYY-MM-DD') + ' até ' + end.format('YYYY-MM-DD'))
  })
```


![image](https://user-images.githubusercontent.com/33382449/64195047-e0784b00-ce56-11e9-8cad-521d3a6019d9.png)

**Comment by dangrossman** [comment]
@felipesilvati You are not having the same problem, you have `firstDay: 7` causing that NaN, because the valid days of the week are numbered 0 through 6

**Comment by dangrossman** [comment]
@ExplosiveGM Yes you need to use the same format everywhere.
